### ISSUE TITLE
HelixJob with multiple workitems with different payloads seem to get last payload

### ISSUE THREAD

**Issue by HaoK** [dotnet/arcade#1245]
We are making some progress in the helix conversion for asp.net, we are generating a payload directory for each of our test projects, and submitting a single helix job with a workitem + payload per test we want to run:

For example: https://mc.dot.net/#/user/HaoK/pr~2Fextensions~2Ftests/test~2Fhaok/11.2.1

The issue I'm seeing right now is that it appears each of the work items is getting sent the payload from the last work item (they are all getting the samples payload (which appears to be the last work item)

For example looking at the run_client.py the contents of payload unzipped are: https://mc.dot.net/#/user/HaoK/pr~2Fextensions~2Ftests/test~2Fhaok/11.2.1/workItem/Microsoft.AspNetCore.Testing.Tests~2Fnet461/wilogs

```
2018-11-02 18:06:11,648: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\4a38cafa-984d-48e1-87b2-9b4d8d9e45fe\Work\33798e4c-c3b9-4e2e-9b96-dda56cf950be\Unzip\Sample.Tests.dll' to 'C:\dotnetbuild\work\4a38cafa-984d-48e1-87b2-9b4d8d9e45fe\Work\33798e4c-c3b9-4e2e-9b96-dda56cf950be\Exec\Sample.Tests.dll'
2018-11-02 18:06:11,648: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\4a38cafa-984d-48e1-87b2-9b4d8d9e45fe\Work\33798e4c-c3b9-4e2e-9b96-dda56cf950be\Unzip\Sample.Tests.dll.config' to 'C:\dotnetbuild\work\4a38cafa-984d-48e1-87b2-9b4d8d9e45fe\Work\33798e4c-c3b9-4e2e-9b96-dda56cf950be\Exec\Sample.Tests.dll.config'
2018-11-02 18:06:11,648: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\4a38cafa-984d-48e1-87b2-9b4d8d9e45fe\Work\33798e4c-c3b9-4e2e-9b96-dda56cf950be\Unzip\Sample.Tests.pdb' to 'C:\dotnetbuild\work\4a38cafa-984d-48e1-87b2-9b4d8d9e45fe\Work\33798e4c-c3b9-4e2e-9b96-dda56cf950be\Exec\Sample.Tests.pdb'
 ```

Here's the work item being sent looking at the build logs which is showing the proper directory for this test:

![image](https://user-images.githubusercontent.com/6537861/47934204-f7938880-de93-11e8-9a37-7f31157c0591.png)

cc @natemcmaster

**Comment by jonfortescue** [comment]
FYI @chcosta @alexperovich

**Comment by chcosta** [comment]
This seems bad.  @alexperovich , where does this fall on your priority list?

FYI @Chrisboh @markwilkie

**Comment by markwilkie** [comment]
@alexperovich is working on this.  @Chrisboh and I need to better fund the effort.  Also, @jonfortescue has cycles starting Monday.

@alexperovich / @jonfortescue - do either of you have suggestions for a workaround or guidance?

**Comment by alexperovich** [comment]
@HaoK can you show the project you are running. A link to the binlog would be useful also.

**Comment by alexperovich** [comment]
Actually, as a quick test. Are all the payload directories you are sending named `publish`. I think the system may be uploading them all to the same container with the name `publish.zip` because the "folder name" of all the directories is the same. So the last one overwrites the others.

**Comment by HaoK** [comment]
Yeah they are all publish because we just using the output of our publish target. I can try copying them to a different folder to see if that will work around this issue.

**Comment by alexperovich** [comment]
Its a one line fix if that is the problem. I can have a PR in a few minutes.

**Comment by HaoK** [comment]
Yeah that was the issue, copying all the payloads to a unique name fixed the overwriting
